src/base.ts

```diff
@@ -0,0 +1,240 @@
+import forEach from 'lodash/forEach';
+import get from 'lodash/get';
+import assign from 'lodash/assign';
+import isPlainObject from 'lodash/isPlainObject';
+import fetch from './fetch';
+import AbortController from './abort-controller';
+import objectToQueryParamString from './object_to_query_param_string';
+import AirtableError from './airtable_error';
+import Table from './table';
+import HttpHeaders from './http_headers';
+import runAction from './run_action';
+import packageVersion from './package_version';
+import exponentialBackoffWithJitter from './exponential_backoff_with_jitter';
+import type Airtable from './airtable';
+
+const userAgent = `Airtable.js/${packageVersion}`;
+
+type BaseRequestOptions = {
+    method?: string;
+    path?: string;
+    qs?: any;
+    headers?: any;
+    body?: any;
+    _numAttempts?: number;
+};
+
+class Base {
+    readonly _airtable: Airtable;
+    readonly _id: string;
+
+    constructor(airtable: Airtable, baseId: string) {
+        this._airtable = airtable;
+        this._id = baseId;
+    }
+
+    table(tableName: string) {
+        return new Table(this, null, tableName);
+    }
+
+    makeRequest(options: BaseRequestOptions) {
+        const that = this;
+
+        options = options || {};
+
+        const method = get(options, 'method', 'GET').toUpperCase();
+
+        const url = `${this._airtable._endpointUrl}/v${this._airtable._apiVersionMajor}/${
+            this._id
+        }${get(options, 'path', '/')}?${objectToQueryParamString(get(options, 'qs', {}))}`;
+
+        const controller = new AbortController();
+
+        const requestOptions: RequestInit = {
+            method,
+            headers: this._getRequestHeaders(get(options, 'headers', {})),
+            signal: controller.signal,
+        };
+
+        if ('body' in options && _canRequestMethodIncludeBody(method)) {
+            requestOptions.body = JSON.stringify(options.body);
+        }
+
+        const timeout = setTimeout(() => {
+            controller.abort();
+        }, this._airtable.requestTimeout);
+
+        return new Promise((resolve, reject) => {
+            fetch(url, requestOptions)
+                .then((resp: Response & {statusCode: Response['status']}) => {
```

Comment:
That syntax is Typescript type syntax. TLDR: Dereferencing a property's type from a type you use square brackets for indexed and keyed properties. A tuple type, `[string, any]` for example would give the type `string` from `[string, any][0]` and `any` from `[string, any][1]`.

If you write `Response.status` you'll get a typescript error.

> Cannot access 'Response.status' because 'Response' is a type, but not a namespace. Did you mean to retrieve the type of the property 'status' in 'Response' with 'Response["status"]'?ts(2713)

I don't know the official definition of a Typescript namespace off the top of my head but it's kind of like an imported module treated as a type. Like with `import * as someFile from './someFile';`, `someFile` would be a namespace iirc.

Reply:
`Response`, like `Request`, are globals in environments that have `fetch` as a global. They're the `Response` and `Request` types that are part of fetch's API.